sktime/transformers/panel/minirocket.py

```diff
@@ -0,0 +1,842 @@
+    """MINIROCKET
+
+    MINImally RandOm Convolutional KErnel Transform
+
+    **Univariate**
+
+    Unviariate input only.  Use class MiniRocketMultivariate for multivariate
+    input.
+
+    @article{dempster_etal_2020,
+      author  = {Dempster, Angus and Schmidt, Daniel F and Webb, Geoffrey I},
+      title   = {{MINIROCKET}: A Very Fast (Almost) Deterministic Transform for
+                 Time Series Classification},
+      year    = {2020},
+      journal = {arXiv:2012.08791}
+    }
+
+    Parameters
+    ----------
+    num_features             : int, number of features (default 10,000)
+    max_dilations_per_kernel : int, maximum number of dilations per kernel (default 32)
+    random_state             : int, random seed (optional, default None)
+    """
+
+    _tags = {"univariate-only": True}
+
+    def __init__(
+        self, num_features=10_000, max_dilations_per_kernel=32, random_state=None
+    ):
+        self.num_features = num_features
+        self.max_dilations_per_kernel = max_dilations_per_kernel
+        self.random_state = (
+            np.int32(random_state) if isinstance(random_state, int) else None
+        )
+        super(MiniRocket, self).__init__()
+
+    def fit(self, X, y=None):
+        """Fits dilations and biases to input time series.
+
+        Parameters
+        ----------
+        X : pandas DataFrame, input time series (sktime format)
+        y : array_like, target values (optional, ignored as irrelevant)
+
+        Returns
+        -------
+        self
+        """
+        X = check_X(X, enforce_univariate=True, coerce_to_numpy=True)
+        X = X[:, 0, :].astype(np.float32)
+        _, n_timepoints = X.shape
+        if n_timepoints < 9:
+            raise ValueError(
+                (
+                    f"n_timepoints must be >= 9, but found {n_timepoints};"
+                    " zero pad shorter series so that n_timepoints == 9"
+                )
+            )
+        self.parameters = _fit(
+            X, self.num_features, self.max_dilations_per_kernel, self.random_state
+        )
+        self._is_fitted = True
+        return self
+
+    def transform(self, X, y=None):
+        """Transforms input time series.
+
+        Parameters
+        ----------
+        X : pandas DataFrame, input time series (sktime format)
+        y : array_like, target values (optional, ignored as irrelevant)
+
+        Returns
+        -------
+        pandas DataFrame, transformed features
+        """
+        self.check_is_fitted()
+        X = check_X(X, enforce_univariate=True, coerce_to_numpy=True)
+        X = X[:, 0, :].astype(np.float32)
+        return pd.DataFrame(_transform(X, self.parameters))
+
+
+@njit(
+    "float32[:](float32[:,:],int32[:],int32[:],float32[:],optional(int32))",
+    fastmath=True,
+    parallel=False,
+)
+def _fit_biases(X, dilations, num_features_per_dilation, quantiles, seed):
+
+    if seed is not None:
+        np.random.seed(seed)
```

Comment:
there's the `check_random_state` function from sklearn which returns an rng object which you can then use via `rng.randn` for example

Reply:
> there's the `check_random_state` function from sklearn which returns an rng object which you can then use via `rng.randn` for example

This is the same problem we have with rocket...  You can't pass a `RandomState` into jit code.  However, unlike with rocket, I think there is a potential workaround here.  We could pass the `RandomState` to `_fit(...)` (regular python), generate a big list of indices using the `RandomState`, and pass the indices into `_fit_biases(...)`.

Reply:
sounds good, also see my comment above on `random_state` in `__init__`